### PR TITLE
install.upgrade: add pre-upgrade sanity check (replaces #891)

### PR DESCRIPTION
**Pull request by jan--f**
Commits from #891 are squashed and rebased onto master. Additionally on commit to pass test suite.

### CODE REVIEW

**Comment by smithfarm** [comment]
@jan--f These are all new commits for teuthology, so I think the "cherry-picked from" lines should be dropped and the resolution descriptions too.

**Comment by jan--f** [comment]
@smithfarm done

**Comment by smithfarm** [comment]
@jan--f Thanks for cleaning up my mess!

File: teuthology/task/install/__init__.py [hunk 1]
```diff
@@ -356,7 +362,7 @@ def upgrade_common(ctx, config, deploy_style):
         pkgs += extra_pkgs
 
         installed_version = packaging.get_package_version(remote, 'ceph-common')
-        upgrade_version = _get_gitbuilder_project(ctx, remote, node).version
+        upgrade_version = get_upgrade_version(ctx, remote, node)
```
Comment (smithfarm) [inline teuthology/task/install/__init__.py]: @jan--f: These improvements seem like they would rather be in the first commit. Would it be a good idea to move them there (and add a `Signed-off-line:` with your name)?

Comment (jan--f) [inline teuthology/task/install/__init__.py]: Hmm I had that thought on another change too. I'll fix it.

**Comment by jan--f** [comment]
@smithfarm ok that looks a bit cleaner

**Comment by jan--f** [comment]
@zmc care to to have a look?

**Comment by smithfarm** [comment]
@dmick @zmc What can I do to get this ball rolling again?

**Comment by dmick** [comment]
first time I'm seeing this, @smithfarm, dunno

**Comment by smithfarm** [comment]
test this please

**Comment by smithfarm** [comment]
@zmc @dmick Any idea why Jenkins appears to be stuck, here?

**Comment by smithfarm** [comment]
Closing in favor of #1087

**Pull request closed without merging**